wp-content/plugins/wporg-learn/inc/post-meta.php

```diff
@@ -257,35 +257,40 @@ function get_workshop_duration( WP_Post $workshop, $format = 'raw' ) {
 }
 
 /**
- * Get a list of locales that are associated with at least one workshop.
- *
- * Optionally only published workshops.
+ * Get a list of locales that are associated with at least one post of the specified type.
  *
  * @param string $meta_key
+ * @param string $post_type
  * @param string $label_language
- * @param bool   $published_only
+ * @param string $post_status
  *
  * @return array
  */
-function get_available_workshop_locales( $meta_key, $label_language = 'english', $published_only = true ) {
+function get_available_post_type_locales( $meta_key, $post_type, $label_language = 'english', $post_status ) {
 	global $wpdb;
 
-	$and_post_status = '';
-	if ( $published_only ) {
-		$and_post_status = "AND posts.post_status = 'publish'";
-	}
-
-	$results = $wpdb->get_col( $wpdb->prepare(
-		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- $and_post_status contains no user input.
-		"
-			SELECT DISTINCT postmeta.meta_value
-			FROM {$wpdb->postmeta} postmeta
-				JOIN {$wpdb->posts} posts ON posts.ID = postmeta.post_id $and_post_status
-			WHERE postmeta.meta_key = %s
-		",
-		$meta_key
-		// phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
-	) );
+	$results = isset( $post_status )
+		? $wpdb->get_col( $wpdb->prepare(
+			"
+				SELECT DISTINCT postmeta.meta_value
+				FROM {$wpdb->postmeta} postmeta
+					JOIN {$wpdb->posts} posts ON posts.ID = postmeta.post_id AND posts.post_type = %s AND posts.post_status = %s 
+				WHERE postmeta.meta_key = %s
+			",
+			$post_type,
+			$post_status,
+			$meta_key
+		) )
+		: $wpdb->get_col( $wpdb->prepare(
+			"
+				SELECT DISTINCT postmeta.meta_value
+				FROM {$wpdb->postmeta} postmeta
+					JOIN {$wpdb->posts} posts ON posts.ID = postmeta.post_id AND posts.post_type = %s 
+				WHERE postmeta.meta_key = %s
+			",
+			$post_type,
+			$meta_key
+		) );
```

Comment:
I worry that duplicating the query here could lead to them accidentally getting out of sync, and it seems less readable to me.

What do you think about integrating `$post_status` into the query conditionally? That seems like the common approach to me. Here's a few examples:

https://github.com/WordPress/wordcamp.org/blob/e3638a232799b31827d4cb107ec503919b0b04ff/public_html/wp-content/plugins/wordcamp-payments-network/includes/payment-requests-list-table.php#L50-L109

https://github.com/WordPress/WordPress/blob/ead3a2ec81f88842c2cba8b8a693871d18522104/wp-includes/canonical.php#L962-L964

https://github.com/WordPress/WordPress/blob/ead3a2ec81f88842c2cba8b8a693871d18522104/wp-includes/class-wp-query.php#L2536-L2537

https://github.com/WordPress/WordPress/blob/ead3a2ec81f88842c2cba8b8a693871d18522104/wp-includes/canonical.php#L940-L950

Reply:
Yeah it's a fair point. I was trying to remove the need for `phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared`, but it seems to be common practice. Changed to something closer resembling the previous code, and updated the comment.